src/Core.hs

```diff
@@ -84,7 +84,7 @@ data CoreF core
   | CoreSyntaxError (SyntaxError core)  -- :: Macro a
   | CoreSendSignal core                 -- :: Signal -> Macro ()
   | CoreWaitSignal core                 -- :: Signal -> Macro ()
-  | CoreIdentEq HowEq core core
+  | CoreIdentEq HowEq core core         -- :: HowEq -> Syntax -> Syntax -> Macro Bool
```

Comment:
The HowEq is insterted by the parser, so the types at runtime are `Syntax -> Syntax -> Macro Bool` instead.